test/test_util.jl

```diff
@@ -87,34 +88,17 @@ end
 Ensure that the gradient w.r.t. the inputs of `MeanFunction` `m` are approximately correct.
 """
 function differentiable_mean_function_tests(
-    m::MeanFunction,
-    ȳ::AbstractVector{<:Real},
-    x::AbstractVector{<:Real};
-    rtol=_rtol,
-    atol=_atol,
+    m::MeanFunction, ȳ::AbstractVector, x::AbstractVector; rtol=_rtol, atol=_atol
 )
     # Run forward tests.
     mean_function_tests(m, x)
 
     # Check adjoint.
     @assert length(ȳ) == length(x)
-    return adjoint_test(x -> ew(m, x), ȳ, x; rtol=rtol, atol=atol)
+    adjoint_test(x -> AbstractGPs._map_meanfunction(m, x), ȳ, x; rtol=rtol, atol=atol)
```

Comment:
```suggestion
    adjoint_test(x -> collect(AbstractGPs._map_meanfunction(m, x)), ȳ, x; rtol=rtol, atol=atol)
```
should address the issues discussed in https://github.com/JuliaGaussianProcesses/AbstractGPs.jl/pull/313/files#r844871805

Reply:
For context: our AD testing infrastructure is really quite reliable when you're working with `Array{Float64, N}`, but is less reliable for other kinds of subtypes of `AbstractArray`. This is obviously really annoying, but we're stuck with it for now.

Reply:
For more context: what is probably going on is that `FiniteDifferences.to_vec(::Fill)` returns not quite the right thing. Ideally we would have better debugging tools there, but they've yet to be written for a number of reasons (available person-time, unclear precisely where some of the rough edges are, etc. The usual reasons)

Reply:
It's also true that we're implicitly assuming that a `Vector{Float64}` is an appropriate tangent type for whatever is returned by `mean(f, x)`. This is definitely not true for `Fill`.

Whichever explanation is correct, I _hope_ that the proposed modification resolves the problem. It very slightly weakens the test, but not by much -- I would certainly be convinced that `mean` is being differentiated correctly if it passes.

Reply:
(Have just run locally -- seems to resolve the problem)